back-end/helpers/get/transcript_data_grouper.spec.js

```diff
@@ -0,0 +1,312 @@
+            const alreadyProcessedIntents = intents.slice(0, 1);
+            const similarIntent = intents[1];
+            const grouper = new TranscriptDataGrouper(alreadyProcessedIntents, 1);
+            
+            const result = await grouper.group(similarIntent);
+            expect(fakeDao.getImmediateIntent).toHaveBeenCalledTimes(2);
+            expect(intents[0].children["h"]).toEqual(33);
+            expect(intents[0].children["c"]).toEqual(33);
+            expect(intents[0].children["b"]).toEqual(33);
+            expect(result).toBe(true);
+        });
+
+        it("should correctly merge 3 similar intents", async() => {
+
+            const alreadyProcessedIntents = intents.slice(0, 1);
+            const grouper = new TranscriptDataGrouper(alreadyProcessedIntents, 1);
+
+            for (const similarIntent of intents.slice(1, 3)) {
+                const result = await grouper.group(similarIntent);
+                expect(result).toBe(true);
+            }
+            expect(fakeDao.getImmediateIntent).toHaveBeenCalledTimes(4);
+            expect(intents[0].children["h"]).toEqual(33);
+            expect(intents[0].children["c"]).toEqual(22);
+            expect(intents[0].children["b"]).toEqual(44);
+        });
+
+        it("should correctly merge first 3 similar intents and not the rest", async() => {
+            const alreadyProcessedIntents = intents.slice(0, 1);
+            const grouper = new TranscriptDataGrouper(alreadyProcessedIntents, 1);
+
+            for (const intent of intents.slice(1)) {
+                await grouper.group(intent);
+            }
+            expect(fakeDao.getImmediateIntent).toHaveBeenCalledTimes(11);
+            expect(intents[0].children["h"]).toEqual(38);
+            expect(intents[0].children["c"]).toEqual(8);
+            expect(intents[0].children["b"]).toEqual(50);
+            expect(intents[0].children["a"]).toEqual(4);
+        });
+        it("should correctly merge 4 similar intents", async() => {
+
+            const alreadyProcessedIntents = intents.slice(0, 1);
+            const grouper = new TranscriptDataGrouper(alreadyProcessedIntents, 1);
+
+            for (const similarIntent of intents.slice(1, 3)) {
+                const result = await grouper.group(similarIntent);
+                expect(result).toBe(true);
+            }
+
+            const lastIntent = intents[6];
+            const result = await grouper.group(lastIntent)
+            expect(fakeDao.getImmediateIntent).toHaveBeenCalledTimes(7);
+            expect(result).toBe(true);
+
+            expect(intents[0].children["h"]).toEqual(40);
+            expect(intents[0].children["c"]).toEqual(13);
+            expect(intents[0].children["b"]).toEqual(40);
+            expect(intents[0].children["a"]).toEqual(7);
+        });
+
+        it("should correctly merge 3 similar intents in different places", async() => {
+            // TODO: Test is currently broken
```

Comment:
Missed todos (from the test being broken from before) to delete

Reply:
good catch